test/lib/independence/independent_request.test.js

```diff
@@ -0,0 +1,146 @@
+const chai = require('chai')
+const sinon = require('sinon')
+const spies = require('chai-spies')
+const proxyquire = require('proxyquire')
+const Promise = require('bluebird')
+
+chai.use(spies)
+
+const mockStorage = {
+  put: (key, value) => { mockStorage._data[key] = value },
+  get: key => mockStorage._data[key],
+  delete: key => { delete mockStorage._data[key] },
+  all: () => mockStorage._data,
+  _data: {}
+}
+const getmockStorage = chai.spy(() => mockStorage)
+const MOCK_DATE = 1558625994627
+
+const { IndependentRequest } = proxyquire('../../../lib/independence/independent_request', {
+  './local_storage_namespace': {
+    LocalStorageNamespace: getmockStorage
+  }
+})
+
+const expect = chai.expect
+
+describe('independent request', () => {
+  let clock
+  const sandbox = chai.spy.sandbox()
+
+  beforeEach(() => {
+    mockStorage._data = { }
+    sandbox.on(mockStorage, ['put', 'get', 'all'])
+    clock = sinon.useFakeTimers(MOCK_DATE)
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+    clock.restore()
+  })
+
+  it('created a storage with the key `independentRequests`', () => {
+    expect(getmockStorage).to.have.been.called.with('independentRequests')
+    expect(IndependentRequest.storage).to.equal(mockStorage)
+  })
+
+  it('defaultly sets the key to be the current date', () => {
+    expect(new IndependentRequest().key).to.equal(MOCK_DATE)
+  })
+
+  it('calls the adapter with the given data', () => {
+    const mockAdapter = chai.spy(() => Promise.resolve({ status: 200 }))
+    const data = { }
+    const request = new IndependentRequest(data)
+
+    return request.send(mockAdapter).then(() => {
+      expect(mockAdapter).to.have.been.called.with(data)
+    })
+  })
+
+  it('saves the request if it fails with status 500', () => {
+    const error = new Error('some error')
+    error.status = 500
+    const mockAdapter = () => Promise.resolve().then(() => { throw error })
```

Comment:
use `Promise.reject(error)`

Reply:
The linter throws an error: `prefer-promise-reject-errors: Expected the Promise reject reason to be an Error.`

Reply:
oh, it's looking for the `new Error` which is fine because we don't really need the error variable.
Also if the eslint have a problem with something that create something weird use a suppress comment:
```js
// eslint-disable-next-line prefer-promise-reject-errors
Promise.reject(error)
...
// or
...
Promise.reject(error) // eslint-disable-line  prefer-promise-reject-errors
```